src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -157,52 +150,18 @@ class SourcesTree extends Component<Props, State> {
       return this.setState({ highlightItems });
     }
 
-    if (nextProps.sources === this.props.sources) {
-      return;
-    }
-
-    if (nextProps.sources.size === 0) {
-      // remove all sources
-      this.setState(
-        createTree(
```

Comment:
i agree it's verbose, but it's nice to see the params

Reply:
Went with the train of thought that for a quick look we only need to know that we create the tree based on the params. If we want to understand what the function does exactly we have to look into the file anyway. Inside the function itself we have a 1 line statement with all the constants we are going to use. So understanding the details of the work of the function felt easier and skimming the file inside which the function is used seemed easier as well.  

No problem changing it back though. Just felt cleaner that way.